src/celeritas/random/distribution/NormalDistribution.hh

```diff
@@ -85,14 +85,15 @@ CELER_FUNCTION auto NormalDistribution<RealType>::operator()(Generator& rng)
     if (has_spare_)
     {
         has_spare_ = false;
-        return spare_ * stddev_ + mean_;
+        return std::fma(spare_, stddev_, mean_);
     }
 
-    real_type theta = 2 * constants::pi * generate_canonical(rng);
-    real_type r = std::sqrt(-2 * std::log(generate_canonical(rng)));
+    constexpr auto twopi = static_cast<real_type>(2 * m_pi);
```

Comment:
We have 9 (now 10) occurrence of `m_pi` and 30 (now 29) of `constants::pi`.  Is there a difference? Do we have guidance of when to use which?

Here is the `constexpr` or the `static_cast` the 'relevant' part?

Either way, if it is usefull, should it be promoted to a global constant?  (of the 30 usage of `constants::pi`, 20 are for `2*`

Reply:
All the `constants` are `real_type`; whereas `m_pi` is `double`. I'll clarify this on the single-precision PR where that's more relevant.